controllers/cluster_controller.go

```diff
@@ -111,6 +111,16 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return ctrl.Result{}, err
 	}
 
+	if cluster.IsManaged() && cluster.Spec.BundlesRef == nil {
+		mgmtCluster := &anywherev1.Cluster{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: cluster.ManagedBy(), Namespace: req.Namespace}, mgmtCluster); err != nil {
+			if apierrors.IsNotFound(err) {
+				return reconcile.Result{}, nil
```

Comment:
why doesn't this return or log the error?

Reply:
Also, why don't we handle the other types of errors?

Reply:
Updated to handle the other errors. We don't log or return the error when the obj isn't found. I'm not sure why but that's the behavior in other places for the controller. It seems pretty intentional so perhaps there are cases where sometimes the objects aren't found but should warrant an error